packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/new-account-form-controller.js

```diff
@@ -409,6 +450,11 @@ export default class NewAccountFormController {
       // update model
       this.model[rule.fieldName] = value;
 
+      // if phone type is set to a value other than 'mobile' we reset the sms consent value
+      if (rule.fieldName === 'phoneType' && value !== 'mobile') {
```

Comment:
It's already inside a ng-change (or the ui-kit equivalent) and if it's there it because the change in one new-account-form-field need to trigger a change in another new-account-form-field